warehouse/query-core/src/main/java/datawave/query/jexl/visitors/IngestTypePruningVisitor.java

```diff
@@ -324,7 +340,7 @@ private Set<String> pruneJunction(JexlNode node, Object data) {
         for (int i = node.jjtGetNumChildren() - 1; i >= 0; i--) {
             node.jjtGetChild(i).jjtAccept(this, data);
         }
-        return Collections.emptySet();
+        return new HashSet<>();
```

Comment:
Is this actually more efficient?

Collections.emptySet() is an immutable object that isn't going to allocate more memory every time that this is called, whereas a new HashSet<>() is going to. Even if it's small, it seems like this could be bad for memory management.

Reply:
The collection returned by `Collections.emptySet()` will throw an UnsupportedOperationException in some very unfortunate circumstances, so I ripped out all Collections and Guava utility methods in favor of "native" collections. Union and intersection operations are now done via `.addAll` and `.retain`

Reply:
SGTM